core/src/main/java/com/rudderstack/android/sdk/core/RudderPreferenceManager.java

```diff
@@ -45,6 +48,18 @@ String getTraits() {
         return preferences.getString(RUDDER_TRAITS_KEY, null);
     }
 
+    String getAnonymousId() {
+        String anonymousId = preferences.getString(RUDDER_ANONYMOUS_ID, null);
+        if (anonymousId == null) {
+            preferences.edit().putString(RUDDER_ANONYMOUS_ID, UUID.randomUUID().toString()).apply();
+        }
+        return preferences.getString(RUDDER_ANONYMOUS_ID, null);
```

Comment:
preferences.apply() is asynchronous. You may not get the correct value from here immediately. Better to return the variable above

Reply:
done